pkg/utils/patch_test.go

```diff
@@ -0,0 +1,194 @@
+package utils
+
+import (
+	"testing"
+
+	crdv1 "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
+)
+
+func TestCreateSnapshotPatch(t *testing.T) {
+	testcases := []struct {
+		name             string
+		originalSnapshot *crdv1.VolumeSnapshot
+		updatedSnapshot  *crdv1.VolumeSnapshot
+
+		expectedPatch string
+	}{
+		{
+			name: "1-1: empty case",
+
+			originalSnapshot: &crdv1.VolumeSnapshot{},
+			updatedSnapshot:  &crdv1.VolumeSnapshot{},
+			expectedPatch:    `{}`,
+		},
+		{
+			name: "1-2 patch new status",
+
+			originalSnapshot: &crdv1.VolumeSnapshot{},
+			updatedSnapshot: &crdv1.VolumeSnapshot{
+				Status: &crdv1.VolumeSnapshotStatus{
+					BoundVolumeSnapshotContentName: stringPtr("test"),
+				},
+			},
+			expectedPatch: `{"status":{"boundVolumeSnapshotContentName":"test"}}`,
+		},
+		{
+			name: "1-3: clear status",
+
+			originalSnapshot: &crdv1.VolumeSnapshot{
+				Status: &crdv1.VolumeSnapshotStatus{
+					BoundVolumeSnapshotContentName: stringPtr("test"),
+					CreationTime:                   nil,
+					ReadyToUse:                     nil,
+					RestoreSize:                    nil,
+					Error:                          nil,
+				},
+			},
+			updatedSnapshot: &crdv1.VolumeSnapshot{
+				Status: &crdv1.VolumeSnapshotStatus{
+					BoundVolumeSnapshotContentName: nil,
+					CreationTime:                   nil,
+					ReadyToUse:                     nil,
+					RestoreSize:                    nil,
+					Error:                          nil,
+				},
+			},
+			expectedPatch: `{"status":{"boundVolumeSnapshotContentName":null}}`,
```

Comment:
Should it be "nil"?

Reply:
The way merge patch works is that it accepts a json object with all of the fields that are being changed. `null` json value marshals to `nil` in Go.

the createSnapshotPatch/createContentPatch funcs create json objects like this.

Reply:
Sure.  Can you add this as comments here in the code?